src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/converter/rulemanager/OfMatchConverter.java

```diff
@@ -32,150 +34,168 @@
+                break;
+            case ETH_DST:
+                builder.field(Field.ETH_DST);
+                builder.value(((MacAddress) ofValueType).getLong());
+                break;
+            case ETH_TYPE:
+                builder.field(Field.ETH_TYPE);
+                builder.value(((EthType) ofValueType).getValue());
+                break;
+            case IP_PROTO:
+                builder.field(Field.IP_PROTO);
+                builder.value(((IpProtocol) ofValueType).getIpProtocolNumber());
+                break;
+            case UDP_SRC:
+                builder.field(Field.UDP_SRC);
+                builder.value(((TransportPort) ofValueType).getPort());
+                break;
+            case UDP_DST:
+                builder.field(Field.UDP_DST);
+                builder.value(((TransportPort) ofValueType).getPort());
+                break;
+            case METADATA:
+                builder.field(Field.METADATA);
+                builder.value(((OFMetadata) ofValueType).getValue().getValue());
+                break;
+            case IN_PORT:
+                builder.field(Field.IN_PORT);
+                builder.value(((OFPort) ofValueType).getPortNumber());
+                break;
+            case VLAN_VID:
+                builder.field(Field.VLAN_VID);
+                builder.value(((OFVlanVidMatch) ofValueType).getVlan());
+                break;
+            case TUNNEL_ID:
+                builder.field(Field.NOVIFLOW_TUNNEL_ID);
+                builder.value(((VxlanNI) ofValueType).getVni());
+                break;
+            case KILDA_VXLAN_VNI:
+                builder.field(Field.OVS_VXLAN_VNI);
+                builder.value(((U64) ofValueType).getValue());
+                break;
+            default:
+                log.debug(String.format("Unexpected match field %s", field.id));
         }
+        return builder.build();
+    }
 
-        // UDP_DST
-        Masked<TransportPort> udpDstMasked = match.getMasked(MatchField.UDP_DST);
-        if (udpDstMasked != null) {
-            fieldMatches.add(FieldMatch.builder()
-                    .field(Field.UDP_DST)
-                    .value(udpDstMasked.getValue().getPort())
-                    .mask((long) udpDstMasked.getMask().getPort())
-                    .build());
-        } else {
-            TransportPort udpDst = match.get(MatchField.UDP_DST);
-            if (udpDst != null) {
-                fieldMatches.add(FieldMatch.builder().field(Field.UDP_DST).value(udpDst.getPort()).build());
-            }
-        }
+    private FieldMatch getMasked(Match match, MatchField field) {
+        FieldMatchBuilder builder = FieldMatch.builder();
+        switch (field.id) {
+            case ETH_SRC:
+                Masked<MacAddress> ethSrcMasked = match.getMasked(MatchField.ETH_SRC);
+                builder.field(Field.ETH_SRC);
+                builder.value(ethSrcMasked.getValue().getLong());
```

Comment:
Value and mask field object have same class and can be handled in same way. It is possible to regroup processing to get rid from mostly same handlers for exacd/masked fields. I am not insist on it, but it would be good.